src/hooks/sponsorships.ts

```diff
@@ -463,22 +442,75 @@ export function useIsEditingSponsorshipFunding(
     )
 }
 
+/**
+ * Returns a callback that takes the user through the process
+ * of editing their stake.
+ */
 export function useEditSponsorshipFunding() {
     const withFlag = useFlagger()
 
-    type params = Parameters<typeof editSponsorshipFunding>
-
     return useCallback(
-        ({ sponsorship, operator }: { sponsorship: params[0]; operator: params[1] }) =>
-            withFlag(
-                flagKey('isEditingSponsorshipFunding', sponsorship.id, operator.id),
-                () => editSponsorshipFunding(sponsorship, operator),
-            ),
+        ({
+            sponsorship,
+            operator,
+        }: {
+            sponsorship: ParsedSponsorship
+            operator: ParsedOperator
+        }) => {
+            void (async () => {
+                try {
+                    try {
+                        await withFlag(
+                            flagKey(
+                                'isEditingSponsorshipFunding',
+                                sponsorship.id,
+                                operator.id,
+                            ),
+                            async () => {
+                                const wallet = await (await getSigner()).getAddress()
+
+                                /**
+                                 * If the sponsorship token info has already been fetched
+                                 * the following call will do nothing. Otherwise, it'll fetch
+                                 * and cache it. Speed things up later on.
+                                 */
+                                await getSponsorshipTokenInfo()
+
+                                const leavePenaltyWei = await getSponsorshipLeavePenalty(
+                                    sponsorship.id,
+                                    operator.id,
+                                )
+
+                                await editStakeModal.pop({
```

Comment:
The Edit stake modal also waits for the Graph to catch up with the indexing to the most relevant block height. It's safe to invalidate queries right after.